webapp/src/components/tutorial/TutorialResetCode.tsx

```diff
@@ -13,19 +13,22 @@ interface TutorialResetCodeProps {
 
 export function TutorialResetCode(props: TutorialResetCodeProps) {
     const { resetTemplateCode, tutorialId, currentStep } = props;
+    const [ reloading, setReloading ] = React.useState(false);
 
-    const onResetClick = () => {
+    const onResetClick = async () => {
         pxt.tickEvent(`tutorial.resetcode`, { tutorial: tutorialId, step: currentStep });
-        resetTemplateCode(true);
+        setReloading(true);
+        await resetTemplateCode(true);
```

Comment:
ah wait question, any chance this throws? if so probably want a try{}finally{}, we've hit annoying infinite loading w/o any indication why before like that~

Reply:
oh yeah good call! have updated